packages/@aws-cdk/aws-lambda/lib/function.ts

```diff
@@ -467,6 +467,29 @@ export class Function extends FunctionBase {
     });
   }
 
+  /**
+   * Add a new version for this Lambda, always with a different name.
+   *
+   * This is similar to the {@link addVersion} method,
+   * but useful when deploying this Lambda through CodePipeline with blue/green deployments.
+   * When using {@link addVersion},
+   * your Alias will not be updated until you change the name passed to {@link addVersion} in your CDK code.
+   * When deploying through a Pipeline,
+   * that might lead to a situation where a change to your Lambda application code will never be activated,
+   * even though it traveled through the entire Pipeline,
+   * because the Alias is still pointing to an old Version.
+   * This method creates a new, unique Version every time the CDK code is executed,
+   * and so prevents that from happening.
+   */
+  public newVersion(): Version {
+    const now = new Date();
+    const year = now.getFullYear();
+    const month = now.getMonth().toString().padStart(2, '0');
+    const day = now.getDate().toString().padStart(2, '0');
+    const seconds = now.getSeconds();
+    return this.addVersion(`${year}${month}${day}-${seconds}`);
```

Comment:
Why not just `now.toISOString()`?

Reply:
`toISOString` has some stupid characters inside of it, but if we don't care that much what the actual name is, sure, I can change it.

Reply:
`now.getSeconds` will return between `0` and `60` (leap second, yeah... it's a thing). So you might issue the same version many, many times.

I'd agree that using `toISOString` is probably safer and also offers a good experience (they're neatly time-ordered :D)

Reply:
Fine fine. Changed to `toISOString`.